app/src/main/java/com/example/harvestfresh/SignupActivity.java

```diff
@@ -0,0 +1,75 @@
+package com.example.harvestfresh;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+public class SignupActivity extends AppCompatActivity {
+
+    private Button btnFinalSignup;
+    private EditText etSignupUsername;
+    private EditText etSignupPassword;
+    private EditText etConfirmPassword;
+    private static final String TAG = "SIGNUPACTIVITY";
```

Comment:
Its generally a good idea to seperate out lists of instantiations like these to make it clear where the private instance end and private static variables begin, so I would put a new line before this. Also, we generally try to sort class variables and methods from most to least important: statics before instance, and public before protected before private, so I would put TAG at the top. Following this convention makes it a lot easier to find code later, and to read through the class later and get a top-down understanding without jumping all over it.